opentelemetry/api/trace/__init__.py

```diff
@@ -0,0 +1,190 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+The OpenTelemetry tracing API describes the classes used to generate
+distributed traces.
+
+The :class:`.Tracer` class controls access to the execution context, and
+manages span creation. Each operation in a trace is represented by a
+:class:`.Span`, which records the start, end time, and metadata associated with
+the operation.
+
+This module provides abstract (i.e. unimplemented) classes required for
+tracing, and a concrete no-op ``BlankSpan`` that allows applications to use the
+API package alone without a supporting implementation.
+
+The tracer supports implicit and explicit context propagation. By default spans
+are created as children of the currently active span, and the newly-created
+span becomes the new active span::
+
+    from opentelemetry.sdk.trace import tracer
+
+    # Create a new root span, set it as the current span in context
+    with tracer.span("parent"):
+        # Attach a new child and update the current span
+        with tracer.span("child"):
+            do_work():
+        # Close child span, set parent as current
+    # Close parent span, set default span as current
+
+Under explicit context propagation there is no concept of an active span, and
+the caller is responsible for managing the span's lifetime::
+
+    from opentelemetry.sdk.trace import tracer
+    from your.integration import deserialize_span
```

Comment:
I think this example does not really match the previous one, with the additional complexity of using a serialized span (though that is certainly a very common use case for explicit context passing). Also, the name `deserialize_span` seems to imply that a whole span is de-serialized. Most likely we would only (de)serialize a span context.

Reply:
+1

Reply:
I'm happy to replace these examples, dealing with a serialized span from another caller was the only example I could think of that someone would use explicit context.

What would you replace this with?

Reply:
What about the scenario where a web request handler schedules an operation in a worker thread (pool)?

Reply:
@Oberon00 depending on whether the handler takes advantage of contextvars or not.
If yes, the contextvars would take care the implicit context.
If not, we might need something like this https://github.com/census-instrumentation/opencensus-python/blob/master/context/opencensus-context/examples/thread_pool.py.